src/main/java/Dog.java

```diff
@@ -23,6 +24,7 @@ public void run(){
     public  void stop(){
         System.out.println(name+" stopped");
         this.run=false;
+        this.walk=false;
     }
 
     public void eat(){
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Rename method "eat" to prevent any misunderstanding/clash with field "eat" defined on line 7. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1845)